### PR TITLE
feat(oas3): add `termsOfService` support

### PR DESCRIPTION
**Pull request by marcofriso**
Closes APIARY-6288

### CODE REVIEW

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js [hunk 1]
```diff
@@ -35,8 +35,17 @@ const isUnsupportedKey = R.anyPass(R.map(hasKey, unsupportedKeys));
...
+    const link = new namespace.elements.Link();
+    link.relation = 'terms-of-service';
+    link.href = terms;
+
+    return link;
+  };
+
   const parseMember = R.cond([
     [hasKey('title'), parseString(context, name, true)],
+    [hasKey('termsOfService'), parseString(context, name, true)],
```
Comment (marcofriso) [inline packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js]: I am not sure about the 3rd parameter, whether should be `true` or `false`

Comment (kylef) [reply packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js]: It should be `false`, it represents that `termsOfService` is optional and not required.

File: packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseInfoObject-test.js [hunk 2]
```diff
@@ -178,4 +165,19 @@ describe('#parseInfoObject', () => {
...
+      termsOfService: 'http://example.com/terms/',
+    });
+
+    const parseResult = parse(context, info);
+    expect(parseResult.length).to.equal(1);
+
+    const termsOfService = parseResult.api.links.get(0);
+    expect(termsOfService.relation.toValue()).to.equal('terms-of-service');
+    expect(termsOfService.href.toValue()).to.equal('http://example.com/terms/');
+  });
```
Comment (kylef) [inline packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseInfoObject-test.js]: Could we get a test to check what happens when the link is not a string?

Something along the lines of:

```patch
+    it('provides a warning when termsOfService is not a string', () => {
+      const info = new namespace.elements.Object({
+        title: 'My API',
+        version: '1.0.0',
+        termsOfService: 6,
+      });
+
+      const parseResult = parse(context, info);
+      expect(parseResult).to.contain.warning("'Info Object' 'termsOfService' is not a string");
+    });
```

I'd wrap these two test cases in `describe('#termsOfService', () => {`

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js [hunk 3]
```diff
@@ -58,6 +67,10 @@ function parseInfo(context, info) {
       api.title = info.get('title');
       api.attributes.set('version', info.get('version'));
 
+      if (info.get('termsOfService')) {
+        api.links.push(parseTermsOfService(info.get('termsOfService')));
```
Comment (kylef) [inline packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js]: The pattern in the code-base is generally in this block of code the everything is already parsed. I'd move the `parseTermsOfService` up a bit for consistency into something like:

```js
  const parseTermsOfService = pipeParseResult(namespace,
    parseString(context, name, false),
    getValue,
    (termsOfService) => {
      const link = new namespace.elements.Link();
      link.relation = 'terms-of-service';
      link.href = termsOfService;
      return link;
    });
```

Then parse the member as: `[hasKey('termsOfService'), parseTermsOfService],` then here it's just about connecting it up:

```js
      if (info.get('termsOfService')) {
        api.links.push(info.get('termsOfService'));
      }
```

File: packages/fury-adapter-oas3-parser/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Fury OAS3 Parser Changelog
 
+## Master
+
+### Enhancements
+
+- Adds support for the `info.termsOfService` schema extension.
```
Comment (kylef) [inline packages/fury-adapter-oas3-parser/CHANGELOG.md]: I'd remove the wording regarding schema and extensions this isn't part of schema object or an extension. Perhaps something along the lines of:

```suggestion
- Adds support for `termsOfService` in 'Info Object'.
```

There's also a file called `STATUS.md` in this directory which tracks what features are supported. Could you take a look at that file and update `termsOfService`.

Comment (kylef) [reply packages/fury-adapter-oas3-parser/CHANGELOG.md]: The STATUS file is at https://github.com/apiaryio/api-elements.js/blob/ed99d4967eb71383e9328cb3c56dcbd74e43262a/packages/fury-adapter-oas3-parser/STATUS.md#info-object

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js (hunk 1 above)
Comment (marcofriso) [reply packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js]: ok, change on next commit

File: packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseInfoObject-test.js (hunk 2 above)
Comment (marcofriso) [reply packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseInfoObject-test.js]: ok, test added in next commit

File: packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js (hunk 3 above)
Comment (marcofriso) [reply packages/fury-adapter-oas3-parser/lib/parser/oas/parseInfoObject.js]: refactor on next commit

File: packages/fury-adapter-oas3-parser/CHANGELOG.md (hunk 4 above)
Comment (marcofriso) [reply packages/fury-adapter-oas3-parser/CHANGELOG.md]: ok, change on next commit

**Pull request merged**